Fresh jokes

mwscanlon opened the pull request:
I was motivated by the jokes plugin scoring an [*F* with codefactor.io](https://www.codefactor.io/repository/github/snoonetirc/cloudbot/file/gonzobot/plugins/jokes.py) to try and clean it up.
I refactored the `load_jokes` function, and renamed some of the global variables that conflicted with the command function names, which I hope bumps it up to an *A*.

While I was at it I went through the other command functions and freshened things up.
Consistent capitalization and punctuation. Some commands had an argument which wasn't in the doc-string. A handful of the functions were using the `message` parameter function when a simple return statement would do. Others redundantly declared a command name that was just the same as the function name.

The only functional changes are that `yomomma` now checks to make sure the target is a valid nick and not the bot itself (otherwise [the caller's nick gets used](https://i.imgur.com/kcLe9rY.gif))
and in `awesome` apparently the wrong link format was being used (I found a link to the source to hopefully aid future maintainers).

mwscanlon committed:
refactor load_jokes

Reduce the complexity by extracting the duplicate with-statements and the line stripping to its own method.

Also switch to using pathlib and the slash operator as done in the 'cheer' plugin.

mwscanlon committed:
define global variables

Fix "Global variable undefined at the module level" warnings by
1. defining them at module level
2. renaming the ones that are already function names

mwscanlon committed:
adjust docstrings

Added missing parameters, consistent capitalisation and punctuation

mwscanlon committed:
spicy momma

Check that the target of yomomma is valid nick and not the bot itself,
otherwise switch things up on the caller

linuxdaemon commented on plugins/jokes.py:
```diff
@@ -1,95 +1,98 @@
 @hook.command(autohelp=False)
-def doit(message):
-    """- prints a do it line, example: mathmaticians do with a pencil"""
-    message(random.choice(do_it))
```
Why replace the call to `message()` here? It would now hilight the user who ran the command where it didn't before.

mwscanlon replied:
In my head I confused the `message` and `reply` parameter-functions so I was under the impression that returning a string called the `message` function anyway. I see now that returning a string calls `reply` which pings the user.
I will undo this…

linuxdaemon commented on plugins/jokes.py:
```diff
@@ -1,95 +1,98 @@
 
 from cloudbot import hook
 
+yo_momma = do_it = puns = confucious_say = one_liner = wise_quotes = book_puns = lawyer_jokes = kero_sayings = ['']
```
It may be cleaner to just use a single global `dict` indexed by the joke type

mwscanlon replied:
Yes, yes

mwscanlon committed:
update awesome command

The format needed of the link appears to have changed.
I included a link I found to the source to hopefully help future
maintainers if it breaks again.

mwscanlon committed:
reduce the number of global variables to one

As per suggestion,
(https://github.com/snoonetIRC/CloudBot/pull/307#discussion_r243883951)
just have one global dict instead of umpteen different variables.

mwscanlon committed:
fix a bug in bookpuns

The logic of the bookpuns function relied on there being only one colon
per line separating the title and the author.
Fun fact, there were two lines with two colons.
If ever a line was added that omitted the colon, that would also break it.

At first I thought to write more careful code for getting the title and
author, but then I realised: why are we reformatting the line every time?
Just write "<title> by <author>" in the data file, boom, solved.

mwscanlon commented:
I also have made changes to the book_puns data file to fix a bug. The code to split the lines into title and author relied on there being only one colon, which is not always the case.

It's a bit unnecessary, I think, to manipulate the text to replace the colon with " by " every time the command is called when that can just be what the data file contains. (I put quotation marks around the title as a personal stylistic choice, but I can change that back or replace with italics formatting very easily.)

linuxdaemon commented on plugins/jokes.py:
```diff
@@ -1,95 +1,99 @@
-
-    with codecs.open(os.path.join(bot.data_dir, "kero.txt"), encoding="utf-8") as f:
-        kero_sayings = [line.strip() for line in f.readlines() if not line.startswith("//")]
+    global joke_lines
```
You only need to declare something global if you are assigning to it, like `joke_lines = {}`. Using `__setitem__` on a dict doesn't require global access to the name, just a reference,

mwscanlon replied:
:heavy_check_mark:

linuxdaemon commented on plugins/jokes.py:
```diff
@@ -1,95 +1,99 @@
+    ]
+    for file_name in file_list:
+        file_path = data_directory / file_name
+        joke_lines[file_path.stem] = load_joke_file(file_path)
```
Perhaps a dynamic system similar to `foods.py` would work well here, as each command behaves pretty much the same. Up to you though, there isn't as much to deal with as there was in `foods.py` so it's not as much of a gain.

mwscanlon replied:
Maybe another time. I've identified a handful of plugins (eightball, fmk, fortune, kenm, topicchange, verysmart) that all have a similar behaviour and combined together it might be worthwhile.

linuxdaemon replied:
yeah, I've had those on my todo list to investigate merging them in to a larger dynamic plugin but haven't gotten around to it.

mwscanlon replied:
I'll assemble a list and create an issue to discuss it.

mwscanlon committed:
remove unnecessary global statement

linuxdaemon commented:
Can you rebase from the `gonzobot` branch just to make sure there aren't any issues?
